Not retrying reader failures

To reproduce:

```
sgload gateload --createreaders --createwriters --numreaders 100 --numwriters 100 --numupdaters 100 --numrevsperdoc 5000 --numdocs 10000 --numchannels 10 --batchsize 100  --expvarprogressenabled --loglevel debug
```

Panic:

```
DBUG[01-03|23:27:16] Statsd timing stat for get_document      delta=5.334162ms
DBUG[01-03|23:27:16] Statsd timing stat for get_document      delta=212.549µs
EROR[01-03|23:27:19] Error calling pullMoreDocs               agent.ID=85 since=1330794-0 err="unexpected EOF"
INFO[01-03|23:27:19] Reader finished                          reader=reader-user-85-fa87112f-41e7-4cc9-bed9-ab1dfe1f0362
EROR[01-03|23:27:19] Error calling pullMoreDocs               agent.ID=2 since=295987-0 err="unexpected EOF"
INFO[01-03|23:27:19] Reader finished                          reader=reader-user-2-fa87112f-41e7-4cc9-bed9-ab1dfe1f0362
INFO[01-03|23:27:19] Reader finished                          agent.ID=2 numdocs=6
panic: Error calling pullMoreDoc: unexpected EOF

goroutine 167 [running]:
panic(0x80eb00, 0xc425f9fe00)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/couchbaselabs/sgload/sgload.(*Reader).Run(0xc420384480)
        /opt/go/src/github.com/couchbaselabs/sgload/sgload/reader.go:126 +0xc82
created by github.com/couchbaselabs/sgload/sgload.GateLoadRunner.startReaders
        /opt/go/src/github.com/couchbaselabs/sgload/sgload/gateload_runner.go:228 +0xff
```

Fix:

Need to log warning and retry instead of panicking.  This was done in other parts of the codebase, but it looks like this particular thing was overlooked

--- a/sgload/reader.go
+++ b/sgload/reader.go
@@ -123,7 +123,7 @@ func (r *Reader) Run() {
 		result, err = r.pullMoreDocs(since)
 		if err != nil {
 			logger.Error("Error calling pullMoreDocs", "agent.ID", r.ID, "since", since, "err", err)
-			panic(fmt.Sprintf("Error calling pullMoreDoc: %v", err))
+			panic(fmt.Sprintf("Error calling pullMoreDocs: %v", err))
 		}
 		since = result.since
 
@@ -223,7 +223,7 @@ func (r *Reader) pullMoreDocs(since Sincer) (pullMoreDocsResult, error) {
 
 	// Create a retry sleeper which controls how many times to retry
 	// and how long to wait in between retries
-	numRetries := 25
+	numRetries := 10
 	sleepMsBetweenRetry := 500
 	retrySleeper := CreateDoublingSleeperFunc(numRetries, sleepMsBetweenRetry)
 
@@ -234,14 +234,16 @@ func (r *Reader) pullMoreDocs(since Sincer) (pullMoreDocsResult, error) {
 
 		changes, newSince, err := r.DataStore.Changes(since, r.BatchSize, r.feedType)
 		if err != nil {
-			return false, err, result
+			logger.Warn("Error getting changes.  Retrying.", "error", err, "agent.ID", r.ID)
+			return true, err, result
 		}
 
 		if len(changes.Results) == 0 {
+			logger.Warn("Got empty changes.  Retrying.", "agent.ID", r.ID)
 			return true, nil, result
 		}
 		if newSince.Equals(since) {
-			logger.Warn("Since value should have changed", "agent.ID", r.ID, "since", since, "newsince", newSince)
+			logger.Warn("Since value has not changed since last _changes request.  Ignorning and retrying.", "agent.ID", r.ID, "since", since, "newsince", newSince, "changes", changes)
 			return true, nil, result
 		}
 

